atomic_reactor/source.py

```diff
@@ -35,21 +35,23 @@
 class SourceConfig(object):
     """ read container.yaml file from build source and store as attrs """
 
-    def __init__(self, build_path):
-        self.data = {}
+    def __init__(self, build_path, provider_params=None):
+        git_params = provider_params or {}
+        self.data = git_params.get('container_yaml', {})
         self.file_path = os.path.join(build_path, REPO_CONTAINER_CONFIG)
-        if os.path.exists(self.file_path):
-            try:
-                # read file and validate against schema
-                self.data = read_yaml_from_file_path(
-                    self.file_path, 'schemas/container.json'
-                ) or {}
-            except Exception:
-                logger.exception(
-                    "Failed to load and validate source config YAML from %s",
-                    self.file_path
-                )
-                raise
+        if not self.data:
+            if os.path.exists(self.file_path):
+                try:
+                    # read file and validate against schema
+                    self.data = read_yaml_from_file_path(
+                        self.file_path, 'schemas/container.json'
+                    ) or {}
```

Comment:
Would prefer this to be part of schema validation of the user params if possible, i.e. including the container.yaml schema.